postreise/plot/plot_pie_generation_vs_capacity.py

```diff
@@ -0,0 +1,248 @@
+                .astype("float")
+                .round(2)
+                .to_dict()
+            )
+            scenario_data[sid]["cap"]["data"][area] = (
+                all_loadzone_data[sid]["cap"][loadzone_set]
+                .sum(axis=1)
+                .divide(1e3)
+                .astype("float")
+                .round(2)
+                .to_dict()
+            )
+    for c_data in custom_data:
+        scenario_data[c_data["name"]] = c_data
+
+    for area in areas:
+        ax_data_list = []
+        for sd in scenario_data.values():
+            for side in ["gen", "cap"]:
+                ax_data, labels = _roll_up_small_pie_wedges(
+                    sd[side]["data"][area], min_percentage
+                )
+
+                ax_data_list.append(
+                    {
+                        "title": "{0}\n{1}".format(sd["name"], sd[side]["label"]),
+                        "labels": labels,
+                        "values": list(ax_data.values()),
+                        "colors": [type2color[r] for r in ax_data.keys()],
+                        "unit": sd[side]["unit"],
+                    }
+                )
+
+        _construct_pie_visuals(area, ax_data_list)
+
+
+def _roll_up_small_pie_wedges(resource_data, min_percentage):
+    """Combine small wedges into a single category. Removes wedges with value 0.
+
+    :param dict resource_data: values for each resource type.
+    :param float min_percentage: roll up small pie pieces into a single category,
+        resources with percentage less than the set value will be pooled together,
+        defaults to 0.
+    :return: (*dict*) -- returns updated axis data and a list of labels that includes
+        the small category label if it exists
+    """
+    resource_list = list(resource_data.keys())
+    total_resources = sum(resource_data.values())
+
+    small_categories = []
+    small_category_value = 0
+    small_category_label = ""
+    for resource in resource_list:
+        percentage = round(resource_data[resource] / total_resources * 100, 1)
+        if percentage == 0.0:
+            resource_data.pop(resource)
+        elif percentage <= min_percentage:
+            small_categories.append(resource)
+            small_category_label += "{0} {1}%\n".format(
+                type2label[resource], percentage
+            )
+            small_category_value += resource_data[resource]
+
+    if len(small_categories) > 1:
+        for resource in small_categories:
+            resource_data.pop(resource)
+
+    labels = [type2label[resource] for resource in resource_data.keys()]
+
+    if len(small_categories) > 1:
+        resource_data["other"] = small_category_value
+        labels.append(small_category_label)
+
+    return resource_data, labels
+
+
+def _construct_pie_visuals(zone, ax_data_list):
+    """Plot formatted data.
+
+    :param str zone: the zone name
+    :param list ax_data_list: a list of labels, values, and colors for each axis of the
```

Comment:
We could mention that the element of the list are dictionaries with keys being labels, values and colors

Reply:
Done.